ocaml/xapi/xapi_ha.ml

```diff
@@ -201,14 +201,17 @@ module Timeouts = struct
   let derive (t : int) =
     (* xHA interface section 4.1.4.1.1 Formula of key timeouts *)
     (* t >= 10 *)
-    if t < 10 then failwith "constraint violation: timeout >= 10" ;
     (* All other values are derived from this single parameter *)
-    let heart_beat_interval = (t + 10) / 10 in
-    let state_file_interval = heart_beat_interval in
+    let round expr = float_of_int t |> expr |> ( +. ) 0.5 |> int_of_float in
+    if t < 10 then failwith "constraint violation: timeout >= 10" ;
+    (* heart beats are cheap but unreliable b/c of UDP - have many *)
+    let heart_beat_interval = 3 in
+    (* state file is slow but realiable  - have 20 *)
+    let state_file_interval = round (fun t -> max 5.0 (t /. 20.0)) in
```

Comment:
As above for small values of `t` this will send less heartbeats making it more likely to fence.
Would it be better to compute the old interval here (`(t+10)/10`)  and take the minimum of the two?

The idea is that we improve reliability if you increase the timeouts, but if you've decreased the timeouts (e.g. because you are confident enought that your network is good and want to failures earlier) then we should not reduce the number of heartbeats we send, otherwise we may end up sending just 1 heartbeat during the entire timeout and if that is lost everything fences.